cadc-dali/src/main/java/ca/nrc/cadc/dali/tables/votable/binary/BinaryRowReader.java

```diff
@@ -0,0 +1,126 @@
+ *  License, or (at your option)         licence, soit (à votre gré)
+ *  any later version.                   toute version ultérieure.
+ *
+ *  OpenCADC is distributed in the       OpenCADC est distribué
+ *  hope that it will be useful,         dans l’espoir qu’il vous
+ *  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+ *  without even the implied             GARANTIE : sans même la garantie
+ *  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+ *  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+ *  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+ *  General Public License for           Générale Publique GNU Affero
+ *  more details.                        pour plus de détails.
+ *
+ *  You should have received             Vous devriez avoir reçu une
+ *  a copy of the GNU Affero             copie de la Licence Générale
+ *  General Public License along         Publique GNU Affero avec
+ *  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+ *  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+ *                                       <http://www.gnu.org/licenses/>.
+ *
+ *  $Revision: 5 $
+ *
+ ************************************************************************
+ */
+
+package ca.nrc.cadc.dali.tables.votable.binary;
+
+import ca.nrc.cadc.dali.tables.votable.VOTableField;
+import ca.nrc.cadc.dali.util.FormatFactory;
+
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/*
+* Reads a single row from a VOTable binary stream.
+* */
+public class BinaryRowReader {
+    private final List<VOTableField> fields;
+    private final FieldProcessorFactory decoderFactory;
+    private final FormatFactory formatFactory = new FormatFactory();
+
+    public BinaryRowReader(List<VOTableField> fields, FieldProcessorFactory decoderFactory) {
+        this.fields = fields;
+        this.decoderFactory = decoderFactory;
+    }
+
+    public List<Object> readRow(DataInputStream in) throws IOException {
+        int numFields = fields.size();
+        int numMaskBytes = (numFields + 7) / 8;
+        byte[] nullMask = new byte[numMaskBytes];
+        int read = in.read(nullMask);
+        if (read < numMaskBytes) {
+            return null; // End of stream
+        }
+
+        List<Object> row = new ArrayList<>(numFields);
+        for (int i = 0; i < numFields; i++) {
+            boolean isNull = (((nullMask[i / 8] & 0xFF) >> (7 - (i % 8))) & 0x01) != 0;
+            if (isNull) {
+                row.add(null);
+            } else {
+                VOTableField field = fields.get(i);
+                boolean isArray = field.getArraysize() != null;
+                int length;
+
+                if (isArray) {
+                    length = field.getArraysize().contains("*") ? in.readInt() : Integer.parseInt(field.getArraysize());
```

Comment:
In the VOTable spec, arraysize can also be used for multi-dimensional arrays, eg
```
arraysize="10x10"
arraysize="10x*"
```
where only the last dimension can be variable. This is extremely rarely used... does this parsing handle that case?